File: SampleObjC/SampleApp/CodeNextViewController.m
#import "CodeNextViewController.h"

@implementation CodeNextViewController

- (instancetype)initWithBackgroundColor:(UIColor *)backgroundColor
{
    self = [super init];
    if (self)
    {
        _backgroundColor = backgroundColor;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.view.backgroundColor = self.backgroundColor;
}

@end

File: SwiftSample/SampleApp/CodeNextViewController.swift
import UIKit

final class CodeNextViewController: UIViewController {
    let backgroundColor: UIColor

    init(backgroundColor: UIColor) {
        self.backgroundColor = backgroundColor
        super.init(nibName: nil, bundle: nil)
    }

    @available(*, unavailable)
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = backgroundColor
    }
}
